AsyncSwift/Managers/Firebase/FirebaseManager.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  FirebaseManager.swift
+//  AsyncSwift
+//
+//  Created by Kim Insub on 2022/11/03.
+//
+
+import Firebase
+import Foundation
+
+final class FirebaseManager: ObservableObject {
+    static let shared = FirebaseManager()
+    let db = Firestore.firestore()
+    private init() { }
+}
+
+extension FirebaseManager {
+    func createUser(user: User) {
+        let docRef = db.collection("users").document(user.id)
+        let docData: [String: Any] = [
+            "id": user.id,
+            "name": user.name,
+            "nickname": user.nickname,
+            "role": user.role,
+            "description": user.description,
+            "linkedInURL": user.linkedInURL,
+            "profileURL": user.profileURL,
+            "friends": []
+        ]
+        docRef.setData(docData) { error in
+            if let error = error {
+                print("Error writing document: \(error)")
+            } else {
+                print("Document successfully written")
+            }
+        }
+    }
+
+    func getUserBy(id: String, completion: @escaping (User) -> Void) {
```

Comment:
함수명이 애매하다고 느껴집니다. 물론 파라미터로 유추할 수 있지만 설명이 없는 함수에 getUserBy는 가독성이 떨어진다고 생각됩니다. 어떤 것으로 User를 알 수 있는지 함수명이 담긴게 좋을 것 같습니다.

Reply:
https://www.swift.org/documentation/api-design-guidelines/#naming

파라미터 네이밍에 함수의 성격을 담는것은 기본적인 애플의 가이드라인입니다 
따라서 문제 없다고 생각하여 리졸브 하도록하겠습니다

Reply:
```suggestion
    func getUser(from id: String, completion: @escaping (User) -> Void) {
```

이게 더 자연스럽지않을까 하는데 어떠세요 ??
제가 익숙한 애플의 Code는 이런 네이밍이었어요